src/preprocessing/particle_packing/system.jl

```diff
@@ -0,0 +1,277 @@
+"""
+    ParticlePackingSystem(shape::Union{InitialCondition, ComplexShape};
+                          smoothing_kernel=SchoenbergCubicSplineKernel{ndims(shape)}(),
+                          smoothing_length=1.2shape.particle_spacing,
+                          signed_distance_field::SignedDistanceField=shape.signed_distance_field,
+                          is_boundary=false, boundary_compress_factor=1.0,
+                          boundary::Union{Polygon, TriangleMesh}=shape.geometry,
+                          neighborhood_search=GridNeighborhoodSearch{ndims(shape)}(),
+                          background_pressure, tlsph=true)
+System to generate body fitted particles for complex shapes.
+For more information about the methods, see description below.
+
+# Arguments
+- `initial_condition`: [`InitialCondition`](@ref) or [`ComplexShape`](@ref) to be packed.
+
+# Keywords
+- `boundary`: Geometry returned by [`load_geometry`](@ref).
+- `background_pressure`: Constant background pressure to physically pack the particles.
+                         A large `background_pressure` can cause high accelerations which requires a properly adjusted time-step criterion.
+- `tlsph`: With the [`TotalLagrangianSPHSystem`](@ref), particles need to be placed
+           on the boundary of the shape and not one particle radius away, as for fluids.
+           When `tlsph=true`, particles will be placed on the boundary of the shape.
+- `is_boundary`: When `is_boundary=true`, boundary particles will be sampled and packed in an offset surface of the `boundary`.
+                 The thickness of the boundary is specified by passing the [`SignedDistanceField`](@ref) of `boundary` with:
+                    - `use_for_boundary_packing=true`
+                    - `max_signed_distance=boundary_thickness`
+- `signed_distance_field`: To constrain particles onto the surface, the information about
+                          the signed distance from a particle to a face is required.
+                          The precalculated signed distances will be interpolated to each particle during the packing procedure.
+- `smoothing_kernel`: Smoothing kernel to be used for this system.
+                      See [Smoothing Kernels](@ref smoothing_kernel).
+- `smoothing_length`: Smoothing length to be used for this system.
+                      See [Smoothing Kernels](@ref smoothing_kernel).
```

Comment:
Please align

Reply:
it is aligned?

Reply:
we have usually also aligned the ":"

Reply:
like so? 
![image](https://github.com/user-attachments/assets/13a20583-0790-40ba-9e1f-16e71c3d55f3)

Reply:
yes